### PR TITLE
[IMP] sale_order_product_recommendation: Get sale orders with sudo to get orders from other salesperson

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT30453

ping @carlosdauden @chienandalu

### CODE REVIEW

File: sale_order_product_recommendation/wizards/sale_order_recommendation.py [hunk 1]
```diff
@@ -63,11 +63,16 @@ def _recomendable_sale_order_lines_domain(self):
...
         )
-        other_sales = self.env["sale.order"].search(
-            [
-                (sale_order_partner_field, "child_of", partner.id),
-                ("date_order", ">=", start),
-            ]
+        # Search with sudo for get sale order from other commercials users
+        other_sales = (
+            self.env["sale.order"]
+            .sudo()
```
Comment (chienandalu) [inline sale_order_product_recommendation/wizards/sale_order_recommendation.py:69]: If the partner is shared between companies, you could get sales from others that the current company, isn't it?

**Comment by sergio-teruel** [comment]
Ok... @chienandalu Changes done!!

**Review by carlosdauden — APPROVED** [review]

File: sale_order_product_recommendation/wizards/sale_order_recommendation.py [hunk 2]
```diff
@@ -63,11 +63,17 @@ def _recomendable_sale_order_lines_domain(self):
...
+            .search(
+                [
+                    ("company_id", "in", self.env.companies.ids),
```
Comment (pedrobaeza) [inline sale_order_product_recommendation/wizards/sale_order_recommendation.py]: Isn't this going to be the order's company?

**Comment by sergio-teruel** [comment]
Changes done!! Now only take into account orders in the sale order company.

**Review by carlosdauden — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Can you please check Travis?

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
There's no cause inside this PR for failing, and we need this fix, so merging manually. I'll put an issue about the current problem.

**Pull request merged**